### PR TITLE
Add config option to set the interface address gNMI should listen on

### LINKED ISSUE #23: Configuration option to listen on only specific interfaces, rather than all of them
The server agent currently starts the gRPC server listening port on all interfaces:
```
    server.add_secure_port('[::]:%d' % gnmi_port, credentials)
```
It would be nice to make that configurable, for cases where you might want to specify as particular interface to listen on, for example `127.0.0.1`.

### PR DESCRIPTION
**Pull request by cglewis**
Closes #23

### CODE REVIEW

File: faucetagent.py [hunk 1]
```diff
@@ -327,12 +327,13 @@ def parse():
...
     arg('--cert', required=True, help='certificate file')
     arg('--key', required=True, help='private key file')
+    arg('--gnmiaddr', default='[::]', help='gNMI address to listen on ([::])')
     arg('--gnmiport', type=int, default=10161, help='gNMI port (10161)')
     arg('--configfile', required=True, help='FAUCET config file')
     arg('--promport',
         type=int,
         default=9302,
-        help='FAUCET prometheus port(9302)')
+        help='FAUCET prometheus port (9302)')
```
Comment (lantz) [inline faucetagent.py]: I'd been meaning to fix this too.

File: faucetagent.py [hunk 2]
```diff
@@ -355,11 +356,12 @@ def main():
     # FaucetAgent handles gNMI requests
     agent = FaucetAgent(proxy)
     # Start the FAUCET gNMI service
-    info('Starting FAUCET gNMI configuration agent on port %d', args.gnmiport)
+    gnmi_url = '{0}:{1}'.format(args.gnmiaddr, args.gnmiport)
+    info('Starting FAUCET gNMI configuration agent on port %s', gnmi_url)
```
Comment (lantz) [inline faucetagent.py]: I think "port" here is inaccurate.

**Comment by lantz** [comment]
Thanks for the PRs btw!!

File: faucetagent.py (hunk 2 above)
Comment (cglewis) [reply faucetagent.py]: Should I just remove port so it just says "agent on ..." ?

**Comment by cglewis** [comment]
@lantz hmm, the rebase somehow included your already merged commit, that's weird.

File: faucetagent.py (hunk 2 above)
Comment (lantz) [reply faucetagent.py]: Works for me.

**Comment by lantz** [comment]
The tricky bit I think is adding a test for this. The current test exercises the default interface, but we could add another test that listens on IPv4 on another loopback address and/or IPv6 and then verifies that `gnmi_capabilities` works properly.

**Comment by lantz** [comment]
> @lantz hmm, the rebase somehow included your already merged commit, that's weird.

Yeah, may have to rebase it again since I just merged another PR.

**Comment by cglewis** [comment]
I think I have a way to add a test for this - take a look at the updated #26 and if that test looks ok, if it does I'll make a similar one for this one that adds an option to test that arg.

**Comment by lantz** [comment]
Sounds good; it looks like the way you did it in #26 is right – add some options and then add some new test methods to `EndToEnd` tests which set the options as desired.

**Comment by lantz** [comment]
Getting the simple stuff right is often tricky:

What do you think are the right command line options for this? For example:
- We could split `--gnmiaddr` and `--gnmiport`.
- We could also require/allow `--gnmiaddr` to include a port. 
- We could also permit `--gnmiport` to specify a listening address

**Comment by lantz** [comment]
I think I'm fine with `--gnmiaddr` including an optional port.
The variable name should probably be`gnmi_addr`.
Would you like to add a test for this?

**Comment by cglewis** [comment]
@lantz updated variable names and added to tests.

File: faucetagent.py [hunk 3]
```diff
@@ -330,12 +330,13 @@ def parse():
     arg = parser.add_argument
     arg('--cert', required=True, help='certificate file')
     arg('--key', required=True, help='private key file')
-    arg('--gnmiport', type=int, default=10161, help='gNMI port (10161)')
+    arg('--gnmi_addr', default='[::]', help='gNMI address to listen on ([::])')
+    arg('--gnmi_port', type=int, default=10161, help='gNMI port (10161)')
```
Comment (lantz) [inline faucetagent.py]: I prefer `--gnmiport` as it's easier to type. I realize that internally we are following the irritating PEP8 convention of using underscores, but that seems to be the FAUCET convention and is also what pylint wants.

**Comment by cglewis** [comment]
Ok, changed.

**Comment by cglewis** [comment]
@lantz it is rebased against master

**Pull request merged**